### PR TITLE
LOG-5384: Drop support for fluentd collector

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Removes support for fluent
* leaves TODO to fix up fluent test that should also apply to vector
* Should simply introduction of new CLF api

### Links
* https://issues.redhat.com/browse/LOG-5384

### CODE REVIEW

File: internal/collector/collector.go [hunk 1]
```diff
@@ -153,7 +138,6 @@ func (f *Factory) NewPodSpec(trustedCABundle *v1.ConfigMap, forwarderSpec loggin
 
 	if f.isDaemonset {
 		podSpec.Volumes = append(podSpec.Volumes,
-			v1.Volume{Name: logContainers, VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: logContainersValue}}},
```
Comment (cahartma) [inline internal/collector/collector.go:156]: Why remove for DS?  Isn't this meant to be for deployment only?

File: test/framework/functional/output_forward.go [hunk 2]
```diff
@@ -95,7 +93,7 @@ func (f *CollectorFunctionalFramework) addForwardOutputWithConf(b *runtime.PodBu
 	}
 
 	log.V(2).Info("Adding container", "name", name)
-	b.AddContainer(name, utils.GetComponentImage(constants.FluentdName)).
+	b.AddContainer(name, "quay.io/openshift-logging/fluentd:5.9").
```
Comment (cahartma) [inline test/framework/functional/output_forward.go:96]: nit, but still used in a few places, so why not add to constants package?   Also, did you remove the `FluentdName` and `FluentdImageEnvVar` from constants file?

File: internal/collector/collector.go (hunk 1 above)
Comment (jcantrill) [reply internal/collector/collector.go:156]: `/var/log/containers` path is deprecated in favor of `/var/log/pods` and has been for a long time.  We have no reason to mount it

File: test/framework/functional/output_forward.go (hunk 2 above)
Comment (jcantrill) [reply test/framework/functional/output_forward.go]: Updated where I could

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**